model/client4.go

```diff
@@ -5005,6 +5005,87 @@ func (c *Client4) PatchChannelModerations(channelID string, patch []*ChannelMode
 	return ChannelModerationsFromJson(r.Body), BuildResponse(r)
 }
 
+func (c *Client4) GetSidebarCategoriesForTeamForUser(userID, teamID, etag string) (*OrderedSidebarCategories, *Response) {
+	route := fmt.Sprintf(c.GetUserRoute(userID) + c.GetTeamRoute(teamID) + "/channels/categories")
```

Comment:
Maybe we can use a similar approaches for other endpoints here and create a method like `c.GetCategoryRoute`.